cdap-common/src/main/java/co/cask/cdap/common/io/Processor.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common.io;
+
+/**
+ * Process input values.
+ *
+ * @param <T> type of the input values
+ */
+public interface Processor<T> {
+
+  /**
+   * Process an input value.
+   *
+   * @param input input value to process
+   */
+  void process(T input);
```

Comment:
Better to have this method has `boolean` type return so that if it return `false`, it means termination of processing (a means for not scanning all files, e.g. one can use it to implements finding a particular file).